pkg/build/registry/buildconfig/rest_test.go

```diff
@@ -289,69 +291,114 @@ func TestUpdateBuildConfigError(t *testing.T) {
 func TestBuildConfigRESTValidatesCreate(t *testing.T) {
 	mockRegistry := test.BuildConfigRegistry{}
 	storage := REST{&mockRegistry}
-	failureCases := map[string]api.BuildConfig{
+	failureCases := map[string]struct {
+		expectSuccess bool
+		data          api.BuildConfig
+	}{
 		"blank sourceURI": {
-			ObjectMeta: kapi.ObjectMeta{Name: "abc"},
-			Parameters: api.BuildParameters{
-				Source: api.BuildSource{
-					Type: api.BuildSourceGit,
-					Git: &api.GitBuildSource{
-						URI: "",
+			false,
+			api.BuildConfig{
+				ObjectMeta: kapi.ObjectMeta{Name: "abc"},
+				Parameters: api.BuildParameters{
+					Source: api.BuildSource{
+						Type: api.BuildSourceGit,
+						Git: &api.GitBuildSource{
+							URI: "",
+						},
 					},
-				},
-				Strategy: api.BuildStrategy{
-					Type: api.STIBuildStrategyType,
-					STIStrategy: &api.STIBuildStrategy{
-						Image: "builder/image",
+					Strategy: api.BuildStrategy{
+						Type: api.STIBuildStrategyType,
+						STIStrategy: &api.STIBuildStrategy{
+							From: kapi.ObjectReference{
+								Name: "builder/image",
+							},
+						},
+					},
+					Output: api.BuildOutput{
+						DockerImageReference: "data/image",
 					},
-				},
-				Output: api.BuildOutput{
-					DockerImageReference: "data/image",
 				},
 			},
 		},
 		"blank DockerImageReference": {
-			ObjectMeta: kapi.ObjectMeta{Name: "abc"},
-			Parameters: api.BuildParameters{
-				Source: api.BuildSource{
-					Type: api.BuildSourceGit,
-					Git: &api.GitBuildSource{
-						URI: "http://github.com/test/source",
+			false,
```

Comment:
this is also invalid because it has no strategy, so it's not necessarily testing the blank dockerimagereference(output) flow.